upup/pkg/kutil/rollingupdate_cluster.go

```diff
@@ -266,8 +290,26 @@ func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval
 			return fmt.Errorf("error deleting instance %q: %v", instanceID, err)
 		}
 
-		// TODO: Wait for node to appear back in k8s
+		// Wait for new EC2 instances to be created
 		time.Sleep(interval)
+
+		// Wait until the cluster is happy
+		// TODO: do we need to respect cloud only??
+		for i := 0; i < retries; i++ {
+
+			_, validateDidNotPass := validate.ValidateCluster(u.Node.ClusterName, instanceGroupList, k8sClient)
```

Comment:
The cluster... The code does not know which node is new... Yay asg. A node can be cordoned, and it is a valid cluster. The new node has to be schedulable.